src/app/pages/swap/swap-container.tsx

```diff
@@ -1,66 +1,128 @@
+      if (currency === Currency.STX) {
+        const balance = allTransferableCryptoAssetBalances.find(
+          x => x.type === 'crypto-currency' && x.blockchain === 'stacks' && x.asset.symbol === 'STX'
+        )!.balance;
+        return { currency, icon: tokenInfo.icon, name: tokenInfo.name, balance };
+      }
+      const balance = allTransferableCryptoAssetBalances.find(
+        x => x.type === 'fungible-token' && alexSDK.getAddressFrom(currency) === x.asset.contractId
+      )?.balance;
+      return {
+        currency,
+        icon: tokenInfo.icon,
+        name: tokenInfo.name,
+        balance: balance ?? createMoney(0, tokenInfo.name, tokenInfo.decimals),
+      };
+    },
+    [allTransferableCryptoAssetBalances]
+  );
 
-  const tempSwapAssetTo: SwapAsset = {
-    balance: createMoney(new BigNumber(0), 'xBTC', 0),
-    icon: XBtcIcon,
-    name: 'Wrapped Bitcoin',
-  };
+  const swappableAssets: SwapAsset[] = useMemo(
+    () => supportedCurrencies.map(getAssetFromAlexCurrency),
+    [getAssetFromAlexCurrency, supportedCurrencies]
+  );
 
-  function onSubmitSwapForReview(values: SwapFormValues) {
-    navigate(RouteUrls.SwapReview, {
-      state: { ...values },
+  const [swapSubmissionData, setSwapSubmissionData] = useState<SwapSubmissionData>();
+  const [slippage, _setSlippage] = useState(0.04);
+
+  async function onSubmitSwapForReview(values: SwapFormValues) {
+    if (values.swapAssetFrom == null || values.swapAssetTo == null) {
+      return;
+    }
+    const [router, lpFee] = await Promise.all([
+      alexSDK.getRouter(values.swapAssetFrom.currency, values.swapAssetTo.currency),
+      alexSDK.getFeeRate(values.swapAssetFrom.currency, values.swapAssetTo.currency),
+    ]);
+    setSwapSubmissionData({
+      swapAmountFrom: values.swapAmountFrom,
+      swapAmountTo: values.swapAmountTo,
+      swapAssetFrom: values.swapAssetFrom,
+      swapAssetTo: values.swapAssetTo,
+      router: router.map(x => getAssetFromAlexCurrency(supportedCurrencies.find(y => y.id === x)!)),
+      liquidityFee: new BigNumber(Number(lpFee)).dividedBy(1e8).toNumber(),
+      slippage,
     });
+    navigate(RouteUrls.SwapReview);
   }
 
-  // TODO: Generate/broadcast transaction > pass real tx data
   function onSubmitSwap() {
+    if (swapSubmissionData == null) {
+      return;
+    }
+    const fromAmount = BigInt(
+      new BigNumber(swapSubmissionData.swapAmountFrom).multipliedBy(1e8).dp(0).toString()
+    );
+    const minToAmount = BigInt(
+      new BigNumber(swapSubmissionData.swapAmountTo)
+        .multipliedBy(1e8)
+        .multipliedBy(1 - slippage)
+        .dp(0)
+        .toString()
+    );
+    const txToBroadcast = alexSDK.runSwap(
+      '', // TODO: current user's stxAddress
+      swapSubmissionData.swapAssetFrom.currency,
+      swapSubmissionData.swapAssetTo.currency,
+      fromAmount,
+      minToAmount,
+      swapSubmissionData.router.map(x => x.currency)
+    );
+    // TODO: broadcast the tx
```

Comment:
Here are two TODO

The `txToBroadcast` contains everything needed to assemble the StackTransaction.

```typescript
export type TxToBroadCast = {
  contractAddress: string;
  contractName: string;
  functionName: string;
  functionArgs: ClarityValue[];
  postConditions: Array<FungiblePostCondition | STXPostCondition>;
};
```

Please help point me at the right place to
1. Read current user's stx address
2. Broadcast this tx.

Reply:
You can use `useCurrentAccountStxAddressState()` for the address. Does `useTransactionBroadcast()` help?